src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useEffect } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { client } from '../../utils/fetchClient';
+import { ErrorMessage } from '../../types/ErrorStatusType';
+
+type TodoItemProps = {
+  todo: Todo;
+  todos?: Todo[] | null;
+  setTodos?: (todos: Todo[]) => void;
+  isTempTodoLoading?: boolean | null;
+  setShowError?: (value: boolean) => void;
+  setErrorMessage?: (errorMessage: ErrorMessage) => void;
+  todoIdsToDelete?: number[];
+  setIsFocusTitleInput?: (value: boolean) => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo,
+  todos,
+  isTempTodoLoading,
+  setTodos,
+  setShowError,
+  setErrorMessage,
+  todoIdsToDelete,
+  setIsFocusTitleInput,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (isTempTodoLoading) {
+      setIsLoading(isTempTodoLoading);
+    }
+  }, [isTempTodoLoading]);
+
+  useEffect(() => {
+    if (todoIdsToDelete && todoIdsToDelete.includes(todo.id)) {
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+    }
+  }, [todoIdsToDelete, todo.id]);
```

Comment:
so many useEffects we should get rid of them
can we handle loading state in App and pass value here
Also temp todo never loading at all 
i cant understand what happened here